pyproject.toml

```diff
@@ -1,261 +1,266 @@
+jsonschema = ">4.18.0"   # Provides validation for YAML plugins
+numexpr = "<2.8.5"         # for efficiency improvements
+psutil = "*"               # Required for memory checks
+pyhdf = "*"                # hdf4 readers (MODIS)
+pyaml-env = "*"            # Reading YAML output config files, with paths
+pyorbital = "*"            # required by satpy
+pypublicdecompwt = "*"     # Required to read SEVIRI data
+rasterio = "*"             # GEOTIFF output
+referencing = "*"
+satpy = "*"               # efficiency improvements >= 0.33.1
+scikit-image = "*"        # Radius based center coverage checks
+
+# Optional dependencies
+# To add a new optional dependency it must be added both here and in one of the "extra"
+# groups below.
+#
+# Doc group
+pydata-sphinx-theme = { version = "*", optional = true }
+sphinx = { version = "<7.2", optional = true }
+sphinx-design = { version = "*", optional = true }
+# Lint group
+bandit = { version = "*", optional = true }
+black = { version = "*", optional = true }
+pylint = { version = "*", optional = true }
+flake8 = { version = "*", optional = true }
+flake8-rst-docstrings = { version = "*", optional = true }
+flake8-docstrings = { version = "*", optional = true }
+flake8-rst = { version = "*", optional = true }
+prettier = { version = "*", optional = true }
+doc8 = { version = "*", optional = true }
+# Test group
+pytest = { version = "*", optional = true }
+pytest-cov = { version = "*", optional = true }
+pixelmatch = { version = "*", optional = true }
+# Debug group
+ipython = { version = "*", optional = true }
+
+#[tool.poetry.dev-dependencies]
+# nothing at this stage;
+
+[tool.poetry.extras]
 doc = [
-    "sphinx<7.2",  # Required for building documentation, v7.2 errors
-    "sphinx-design",  # Required for building documentation
-    "pydata-sphinx-theme",  # Required for building documentation
+    "pydata-sphinx-theme", # Required for building documentation
+    "sphinx",              # Required for building documentation, v7.2 errors
+    "sphinx-design"        # Required for building documentation
 ]
 lint = [
-    "bandit",  # Syntax/security checking
-    "black",   # Syntax checking and reformatting
-    "pylint",  # Syntax checking
-    "flake8",  # Syntax checking
-    "flake8-rst-docstrings",  # numpy docstring checking
-    "flake8-docstrings",  # docstring checking
-    "flake8-rst",  # RST checking
-    "prettier",  # YAML, JSON, etc linting
-    "doc8",  # RST linting
+    "bandit",                # Syntax/security checking
+    "black",                 # Syntax checking and reformatting
+    "pylint",                # Syntax checking
+    "flake8",                # Syntax checking
+    "flake8-rst-docstrings", # numpy docstring checking
+    "flake8-docstrings",     # docstring checking
+    "flake8-rst",            # RST checking
+    "prettier",              # YAML, JSON, etc linting
+    "doc8"                   # RST linting
+]
```

Comment:
Are all of the dependencies specified twice?  under lint = [, and with the {} above?  @jsolbrig

Reply:
Yes, yes they are... They have to be specified twice. The first time specifies which version to use and that it is optional. The second specifies which "extras" include each package.